### PR TITLE
Modify CupertinoSearchTextField's prefix icon.

### LINKED ISSUE #80891: Add support for custom prefixIcon in CupertinoSearchTextField
## Use case

I would like to be able to add a custom prefixIcon to the [CupertinoSearchTextField](https://api.flutter.dev/flutter/cupertino/CupertinoSearchTextField-class.html)

## Proposal

Currently `CupertinoSearchTextField` defaults to `CupertinoIcons.search`

custom

`CupertinoSearchTextField` should accept any prefixIcon Widget and default to `CupertinoIcons.search`

cupertino

### PR DESCRIPTION
**Pull request by rickimaru18**
Added "prefixIcon" property so that users can modify its behavior.

Fixes: #80891

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by pedromassango — CHANGES_REQUESTED** [review]
This PR needs test to make sure this works as expected and won't break with future changes.

**Comment by VictorUvarov** [comment]
https://github.com/flutter/flutter/blob/master/packages/flutter/test/cupertino/search_field_test.dart Here are the search field tests they currently have. You can probably just test if the custom prefix icon is visible when overriding.

**Comment by rickimaru18** [comment]
> You can probably just test if the custom prefix icon is visible when overriding.

@VictorUvarov Thanks for the suggestion. 🙇

File: packages/flutter/test/cupertino/search_field_test.dart [hunk 1]
```diff
@@ -259,6 +259,38 @@ void main() {
     },
   );
 
+  testWidgets(
+    'prefix widget visibility',
```
Comment (pedromassango) [inline packages/flutter/test/cupertino/search_field_test.dart:263]: Those can be a one line

File: packages/flutter/lib/src/cupertino/search_field.dart [hunk 2]
```diff
@@ -387,7 +393,7 @@ class _CupertinoSearchTextFieldState extends State<CupertinoSearchTextField>
     final Widget prefix = Padding(
       child: IconTheme(
         data: iconThemeData,
-        child: const Icon(CupertinoIcons.search),
+        child: widget.prefixIcon,
```
Comment (pedromassango) [inline packages/flutter/lib/src/cupertino/search_field.dart:396]: I thing you need to update the docus of the `enabled` property to make it clear the the `prefixIcon` will not respond to touch events.

See https://github.com/flutter/flutter/pull/79966#discussion_r610371491 for context.
 Cc @LongCatIsLooong

**Review by pedromassango — CHANGES_REQUESTED** [review]
Just need some docs update and code format and we good to go.

File: packages/flutter/lib/src/cupertino/search_field.dart (hunk 2 above)
Comment (rickimaru18) [reply packages/flutter/lib/src/cupertino/search_field.dart:399]: I also updated the documentation of `CupertinoSearchTextField`.

**Review by pedromassango — APPROVED** [review]
LGTM

**Comment by pedromassango** [comment]
Failing tests does not seems to be related to this PR.

File: packages/flutter/lib/src/cupertino/search_field.dart [hunk 3]
```diff
@@ -246,6 +250,11 @@ class CupertinoSearchTextField extends StatefulWidget {
   /// the comparison tool in https://github.com/flutter/platform_tests/.
   final EdgeInsetsGeometry prefixInsets;
 
+  /// Sets a prefix widget.
+  ///
+  /// Cannot be null. Defaults to [CupertinoIcons.search].
```
Comment (LongCatIsLooong) [inline packages/flutter/lib/src/cupertino/search_field.dart:255]: nit: `[CupertinoIcons.search]` is not a widget.

**Review by LongCatIsLooong — APPROVED** [review]
LGTM with nits.

File: packages/flutter/lib/src/cupertino/search_field.dart (hunk 3 above)
Comment (rickimaru18) [reply packages/flutter/lib/src/cupertino/search_field.dart:255]: Understood 🙇

Comment (rickimaru18) [reply packages/flutter/lib/src/cupertino/search_field.dart]: @LongCatIsLooong Is this okay? a7ace8e

File: packages/flutter/lib/src/cupertino/search_field.dart [hunk 4]
```diff
@@ -246,6 +250,11 @@ class CupertinoSearchTextField extends StatefulWidget {
   /// the comparison tool in https://github.com/flutter/platform_tests/.
   final EdgeInsetsGeometry prefixInsets;
 
+  /// Sets a prefix widget.
+  ///
+  /// Cannot be null. Defaults to [Icon] with [IconData] [CupertinoIcons.search].
```
Comment (LongCatIsLooong) [inline packages/flutter/lib/src/cupertino/search_field.dart:255]: maybe

> Cannot be null. Defaults to an [Icon] widget with the [CupertinoIcons.search] icon

?

Comment (rickimaru18) [reply packages/flutter/lib/src/cupertino/search_field.dart]: Done 🙇

**Comment by LongCatIsLooong** [comment]
@rickimaru18 Thanks for the quick update! The CI is failing but it does not look related. Could you try merging the lastest `upstream/master` branch and see if that fixes the test failures?

**Pull request merged**